### PR TITLE
Rick and Morty - JUANLUIS

### PR DESCRIPTION
**Pull request by Juanluiscisnerosaldrete**
Page web the rick and morty created, integration of the rick and morty api.
image
image

### CODE REVIEW

File: juanluis/rickandmorty/index.html [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+</head>
+
+<header>
+    <div class="menu-container">
+        <div class="menu-izq">
+            <div class="menu-container1"><a href="#"><img
+                        src="https://i.pinimg.com/550x/b5/c6/39/b5c6399f6ebc60ca27a538a3cd2ed099.jpg" alt=""></a></div>
+        </div>
+        <div class="menu-der">
```
Comment (RayVazquez) [inline juanluis/rickandmorty/index.html:19]: menu-izq and menu-der in english

File: juanluis/rickandmorty/index.html [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    <title>Rick and Morty</title>
+    <link rel="icon" type="image/x-icon" href="https://static-00.iconduck.com/assets.00/rick-icon-512x496-zpvsqbz6.png">
+    <link rel="stylesheet" href="styles.css">
+
+</head>
+
+<header>
+    <div class="menu-container">
+        <div class="menu-izq">
+            <div class="menu-container1"><a href="#"><img
```
Comment (RayVazquez) [inline juanluis/rickandmorty/index.html:16]: be more specific in your class

File: juanluis/rickandmorty/index.html [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    </div>
+</header>
+
+<body>
+    <div class="mycontainer"></div>
+
+    <div class="footer">
+        <div class="foo"><a class="active" href="#">CHARACTERS: 826</a></div>
+        <div class="foo"><a class="active" href="#">LOCATIONS: 126</a></div>
+        <div class="foo"><a class="active" href="#">EPISODES: 51</a></div>
```
Comment (RayVazquez) [inline juanluis/rickandmorty/index.html:36]: what foo means?

File: juanluis/rickandmorty/scripts.js [hunk 4]
```diff
@@ -0,0 +1,54 @@
+let url = "https://rickandmortyapi.com/api/";
+let html = document.querySelector(".mycontainer");
+
+let arrayper = [];
+
+// Genera un array con 6 números aleatorios entre 1 y 100
+for (let i = 0; i < 6; i++) {
+  arrayper.push(Math.floor(Math.random() * (826 - 1) + 1));
```
Comment (RayVazquez) [inline juanluis/rickandmorty/scripts.js:8]: what does this line do?

File: juanluis/rickandmorty/scripts.js [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+    });
+};
+
+// Función principal para obtener datos de personajes y episodios
+let obtenerDatos = (apiURL) => {
+  return fetch(`${apiURL}character/${arrayper}`)
+    .then((res) => res.json())
+    .then(async (response) => {
+      // Utiliza Promise.all para hacer solicitudes concurrentes a la API de episodios
+      const tpt = await Promise.all(
```
Comment (RayVazquez) [inline juanluis/rickandmorty/scripts.js:26]: variable names more specific

File: juanluis/rickandmorty/scripts.js [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+        response.map(async (character) => {
+          // Obtiene el nombre del episodio utilizando la función anterior
+          const episodeName = await obtenerEpisodioNombre(character.episode[0]);
+          // Construye el HTML para cada personaje con su respectivo episodio
+          return `<div class="sub-box">
+                        <img src="${character.image}" id="img" alt="">
+                        <div class="text-container">
+                            <div class="tituloper"><a href="#"><strong>${character.name}</strong></a></div>
```
Comment (RayVazquez) [inline juanluis/rickandmorty/scripts.js:34]: class name in english

File: juanluis/rickandmorty/scripts.js [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+                            <div class="status">
+                                <div class="circle"></div>
+                                <p class="inftitulo">${character.status} - ${character.species}</p>
```
Comment (RayVazquez) [inline juanluis/rickandmorty/scripts.js:37]: english

File: juanluis/rickandmorty/scripts.js [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+                            </div>
+                            <p class="caracter1">Last known location:</p>
+                            <div class="infcaracter"><a href="#">${character.location.name}</a></div>
+
+                            <p class="caracter2">First seen in:</p>
```
Comment (RayVazquez) [inline juanluis/rickandmorty/scripts.js:42]: caracter1 and caracter2 more specific

File: juanluis/rickandmorty/styles.css [hunk 9]
```diff
@@ -0,0 +1,210 @@
...
+    margin-bottom: 15px;
+}
+
+.tituloper a {
+    display: flex;
+    color: rgb(255, 255, 255);
+    text-align: center;
+    text-decoration: none;
+    font-size: 20px;
+}
```
Comment (RayVazquez) [inline juanluis/rickandmorty/styles.css:100]: .tituloper a and .text-container . infcaracter a share some properties, make one global class

File: juanluis/rickandmorty/styles.css [hunk 10]
```diff
@@ -0,0 +1,210 @@
...
+.text-container .inftitulo {
+    font-size: 15px;
+    margin-bottom: 10px;
+}
+
+.text-container .caracter1,
+.text-container .caracter2 {
+    font-size: 15px;
+    color: grey;
+}
```
Comment (RayVazquez) [inline juanluis/rickandmorty/styles.css:111]: why use two classes with the same purpose?

File: juanluis/rickandmorty/styles.css [hunk 11]
```diff
@@ -0,0 +1,210 @@
...
+
+.tituloper a:hover,
+.infcaracter a:hover {
+    color: darkorange;
+}
```
Comment (RayVazquez) [inline juanluis/rickandmorty/styles.css:116]: group all hover elements with the same text color in one line

File: juanluis/rickandmorty/index.html (hunk 1 above)
Comment (Juanluiscisnerosaldrete) [reply juanluis/rickandmorty/index.html:19]: Complete.

File: juanluis/rickandmorty/index.html (hunk 2 above)
Comment (Juanluiscisnerosaldrete) [reply juanluis/rickandmorty/index.html:16]: Complete.

File: juanluis/rickandmorty/index.html (hunk 3 above)
Comment (Juanluiscisnerosaldrete) [reply juanluis/rickandmorty/index.html:36]: Jajaja complete.

File: juanluis/rickandmorty/index.html [hunk 12]
```diff
@@ -0,0 +1,54 @@
...
+    </div>
+
+    <div class="footer2"><img src="netlify.png" alt=""><img
+            src="https://upload.wikimedia.org/wikipedia/commons/9/97/Netlify_logo_%282%29.svg" alt=""></div>
+    <div class="footer3"><img src="pngegg.png" alt=""><img
+            src="https://www.pngfind.com/pngs/b/39-398575_free-png-download-black-twitter-logo-without-white.png"
+            alt=""><img src="https://www.dlf.pt/png/big/17/171073_corazon-blanco-png.png" alt=""></div>
+    <div class="footer1">
+        <>by<a class="active" href="#">JUAN LUIS</a>2024
+    </div>
```
[the remark this answers is not in the document]
Comment (Juanluiscisnerosaldrete) [reply juanluis/rickandmorty/index.html:50]: ready, completed.

File: juanluis/rickandmorty/scripts.js (hunk 4 above)
Comment (Juanluiscisnerosaldrete) [reply juanluis/rickandmorty/scripts.js:8]: Explained and finished.

File: juanluis/rickandmorty/index.html [hunk 13]
```diff
@@ -28,24 +26,24 @@ <h1><strong>The Rick and Morty API</strong></h1>
...
-    <div class="mycontainer"></div>
+    <div class="main-container"></div>
 
     <div class="footer">
-        <div class="foo"><a class="active" href="#">CHARACTERS: 826</a></div>
-        <div class="foo"><a class="active" href="#">LOCATIONS: 126</a></div>
-        <div class="foo"><a class="active" href="#">EPISODES: 51</a></div>
+        <div class="sub-footer"><a class="active" href="#">CHARACTERS: 826</a></div>
+        <div class="sub-footer"><a class="active" href="#">LOCATIONS: 126</a></div>
+        <div class="sub-footer"><a class="active" href="#">EPISODES: 51</a></div>
```
Comment (RayVazquez) [inline juanluis/rickandmorty/index.html:34]: This information should be also consumed by the API

File: juanluis/rickandmorty/scripts.js [hunk 14]
```diff
@@ -1,46 +1,49 @@
...
     .then((res) => res.json())
     .then((episodeData) => {
-      return episodeData.name;
+      return episodeData.name; //Aqui me recupera y me retorna el nombre del episodio
     });
 };
 
 // Función principal para obtener datos de personajes y episodios
-let obtenerDatos = (apiURL) => {
+let getDatos = (apiURL) => {
```
Comment (RayVazquez) [inline juanluis/rickandmorty/scripts.js:22]: getDatos -> getData

File: juanluis/rickandmorty/scripts.js (hunk 5 above)
Comment (RayVazquez) [reply juanluis/rickandmorty/scripts.js:28]: change this variable name

**Pull request merged**